packages/flutter/test/material/navigation_rail_test.dart

```diff
@@ -2022,6 +2022,80 @@ void main() {
 
     semantics.dispose();
   });
+
+  testWidgets('NavigationRailDestination padding properly applied', (WidgetTester tester) async {
+    const EdgeInsets firstItemPadding = EdgeInsets.all(20);
+    const EdgeInsets secondItemPadding = EdgeInsets.symmetric(vertical: 30);
+    const EdgeInsets thirdItemPadding = EdgeInsets.symmetric(horizontal: 10);
+
+    await _pumpNavigationRail(
+      tester,
+      navigationRail: NavigationRail(
+        labelType: NavigationRailLabelType.all,
+        selectedIndex: 0,
+        destinations: const <NavigationRailDestination>[
+          NavigationRailDestination(
+            icon: Icon(Icons.favorite_border),
+            selectedIcon: Icon(Icons.favorite),
+            label: Text('Abc'),
+            padding: firstItemPadding,
+          ),
+          NavigationRailDestination(
+            icon: Icon(Icons.bookmark_border),
+            selectedIcon: Icon(Icons.bookmark),
+            label: Text('Def'),
+            padding: secondItemPadding,
+          ),
+          NavigationRailDestination(
+            icon: Icon(Icons.star_border),
+            selectedIcon: Icon(Icons.star),
+            label: Text('Ghi'),
+            padding: thirdItemPadding,
+          ),
+        ],
+      ),
+    );
+
+    final Padding firstItem = tester.widget<Padding>(find.widgetWithText(Padding, 'Abc').last);
+    final Padding secondItem = tester.widget<Padding>(find.widgetWithText(Padding, 'Def').last);
+    final Padding thirdItem = tester.widget<Padding>(find.widgetWithText(Padding, 'Ghi').last);
+
+    expect(firstItem.padding, firstItemPadding);
+    expect(secondItem.padding, secondItemPadding);
+    expect(thirdItem.padding, thirdItemPadding);
+  });
+
+  testWidgets('No padding is applied by default', (WidgetTester tester) async {
+    // The default padding applied around the content of each item.
```

Comment:
Actually, looking at the [code](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/navigation_rail.dart#L716), some default padding is added to some types of navigation rail destinations (as this test points out). Wouldn't it be better to just modify this default value for types that have padding and also add a padding for the `none` case?

cc/ @clocksmith

Reply:
@shihaohong yes, that would be better I guess. Should I update the PR with this suggestion?

Reply:
I guess for the `none` case, you could just wrap the `content` with a `Padding` widget instead since it seems like there is no default padding. Then, you would then just modify the `Container`s for all other types to use the passed in padding, or the default padding if the passed in padding is null.

Reply:
I would play around with each type in a Flutter app to make sure they all appear okay

Reply:
Made the changes :+1: